File: swift-函数/TDProject/TDProject/Pucharse/Controller/ViewController.m
//
//  ViewController.m
//  TDProject
//
//  Created by mac on 2019/4/29.
//  Copyright © 2019年 mac. All rights reserved.
//

#import "ViewController.h"
#import "PurchaseCell.h"

@interface ViewController ()<UITableViewDelegate,UITableViewDataSource>
@property (nonatomic, strong) UITableView *tableView;
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    [self setUpTableView];
}

- (void)setUpTableView{
    [self registerNib];
    [self.view addSubview:self.tableView];
}

- (void)registerNib{
    [self.tableView registerNib:[UINib nibWithNibName:@"PurchaseCell" bundle:nil] forCellReuseIdentifier:@"PurchaseCell"];
}

- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView{
    return 1;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section{
    return 5;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath{
    PurchaseCell *cell = [tableView dequeueReusableCellWithIdentifier:@"PurchaseCell"];
    cell.productNameLabel.text = [NSString stringWithFormat:@"product %ld",indexPath.row];
    return cell;
    
}

#pragma -mark lazy
- (UITableView *)tableView{
    if (!_tableView) {
        _tableView = [[UITableView alloc] initWithFrame:self.view.bounds style:UITableViewStylePlain];
        _tableView.delegate = self;
        _tableView.dataSource = self;
    }
    return _tableView;
}

@end

File: swift_1/tableView_swift/tableView_swift/ViewController.swift
//
//  ViewController.swift
//  tableView_swift
//
//  Created by mac on 2019/4/21.
//  Copyright © 2019年 mac. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    lazy var tableview : UITableView = UITableView()
    override func viewDidLoad() {
        super.viewDidLoad()
        setUpUI()
    }
}

extension ViewController{
    func setUpUI(){
        tableview.frame = self.view.bounds
        view.addSubview(tableview)
        tableview.delegate = self
        tableview.dataSource = self
    }
}

extension ViewController : UITableViewDelegate,UITableViewDataSource{
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return 20
    }
    
    func numberOfSections(in tableView: UITableView) -> Int {
        return 1
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let id : String = "cellID"
        var cell = tableview.dequeueReusableCell(withIdentifier: id) as? HomeCell
        if cell == nil {
            cell = HomeCell(style: .default, reuseIdentifier: id)
        }
        
        cell?.label1?.text = "label = \(indexPath.row)"
        cell?.label2?.text = "label2 = \(indexPath.row)"
        cell?.label3?.text = "label3 = \(indexPath.row)"
        
        return cell!
    }
}



